clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -9933,6 +9933,21 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
         TCArgs.MakeArgString("--out-file-list=" + OutputFileName));
     ForeachArgs.push_back(
         TCArgs.MakeArgString("--out-replace=" + OutputFileName));
+    // If fsycl-dump-device-code is passed, put the output files from llvm-spirv
```

Comment:
`SPIRVTranslator::ConstructJob` seems to construct the `llvm-foreach` and `llvm-spirv` command, that generates the `.spv` files.
`llvm-foreach` has a `--out-dir` option using which users can specify the directory for output files; If unspecified,           system temporary directory is assumed.

I tried adding this option `--out-dir=/test`, and the .spv files got put in the test directory.

In the code below, I'm trying to get the path value provided in fsycl-dump-device-code and set is as the `--out-dir` value 
Is this approach ok?

Reply:
Can you comment on this?

Reply:
Sure, this is fine for redirecting the produced output from the translator into a specific directory.